src/nanoemoji/write_font.py

```diff
@@ -356,6 +361,13 @@ def _ufo_colr_layers(colr_version, colors, color_glyph, glyph_cache):
         else:
             glyph = glyph_cache[glyph_cache_key]
 
+        glyph_bbox = glyph.getControlBounds(color_glyph.ufo)
```

Comment:
Optional, alternately compute bounds on ColorGlyph alongside of computing the painted layers so here you just access, e.g. color_glyph.bbox

Reply:
Not quite sure what glyph.getControlBounds does ... how is this accounting for paint me that glyph with that transform?

Reply:
We can't compute it on the painted layers, those still contain svg paths. We have to compute the bounds on the layer glyph coordinates after they have been mapped to UPEM space. It's also easier to do it on a UFO glyph with the getControlBounds method than on the svg paths. the `ufoLib2.Glyph.getControlBounds` method uses the fonttools ControlBoundsPen to return the bounding box for the control points of that glyph. It's `None` when the glyph is empty and has no contours nor components. The reason you need to pass the glyphSet parameter to getControlBounds is to be able to resolve component references, as the glyph object does not have a reference to the layer it is in (I'm passing the ufo object itself as the glyphSet for resolving component references, that's equivalent of passing the UFO default layer)

Reply:
> We have to compute the bounds on the layer glyph coordinates after they have been mapped to UPEM space and user-transformed

Or compute the bbox on the layers then map it into upem space?

> return the bounding box for the control points of that glyph

How are we accounting for the DAG's impact on the glyph bounds? For example, if my graph says PaintScaled(double size, center 100,100) > PaintGlyph(a box from 100,100 to 200,200) I would expect glyph control bounds to be 100,100 => 200,200 (not accounting for the scale).

Reply:
> How are we accounting for the DAG's impact on the glyph bounds? 

damn. We aren't. Need rethink

Reply:
> compute the bbox on the layers then map it into upem space?

i found it much easier to reuse ready made method for computing bbox of UFO glyphs, rather than work on svg paths

Reply:
> if my graph says PaintScaled...

FWIW note that right now we aren't using any of that